boot/freeldr/freeldr/lib/fs/fat.c

```diff
@@ -1557,7 +1520,40 @@ ARC_STATUS FatSeek(ULONG FileId, LARGE_INTEGER* Position, SEEKMODE SeekMode)
     if (Position->LowPart >= FileHandle->FileSize)
         return EINVAL;
 
+    {
```

Comment:
Can you move that into a helper function?
Also I was wondering whether these calculations could be deferred in the functions where the CurrentCluster is needed?

Reply:
> Can you move that into a helper function?

Why do you think it would be better to move that into a helper function? I don't use this code anywhere else.

> I was wondering whether these calculations could be deferred

I think that will make the driver more complicated for no reason. We may just assume that ArcSeek is a somehow expensive operation when using it in FreeLoader and not to make useless calls to it
BTW, it will return an error if the FAT is corrupted - you will notice the error earlier